honeybee_vtk/assistant.py

```diff
@@ -20,16 +20,19 @@ class Assistant:
             color that you'd like to set as the background color.
         camera: A Camera object.
         actors: A dictionary of actors from a Scene object
+        legend_parameters: A list of legend parameter objects to be added to the scene
     """
+
     def __init__(self, background_color: Tuple[int, int, int], camera: Camera,
-                 actors: dict) -> None:
+                 actors: dict, legend_parameters: list) -> None:
 
         self._background_color = background_color
         self._actors = actors
         self._camera = camera
         self._interactor = None
         self._window = None
         self._renderer = None
+        self._legend_params = legend_parameters
```

Comment:
What are these legend parameters for? Are they meaningful detached from data?

Reply:
To follow VTK's filmset anology, "An assistant readies everything for a perfect shot."

The scene module has a method named update_scene. In that method, an assitant collects the latest actors, cameras and legends setup in the scene. An assistant object also export images of a scene object and we need to bring all the information we need in the image to the assistant object. Therefore, you find legend_params here.

Please note that these parameters are still tied to data. As per our agreement, the origin of legend_params is DataFieldInfo object.

Reply:
> Please note that these parameters are still tied to the data. As per our agreement, the origin of legend_params is DataFieldInfo object.

Shouldn't this be a property then? It shouldn't be set in the assistant class. Do you have an example that I can check to see how is this useful?

Reply:
> Shouldn't this be a property then?
Property of an assistant object? I don't think so. Because, no one will use that property. [See](https://github.com/devngc/honeybee-vtk/blob/feature-legend/honeybee_vtk/scene.py#L133) how assistants are assigned to a scene. I should say that an assistant object is not a public object. It's used internally to collect actors, cameras and legend_parameters.

Reply:
If it's an internal object then I'm fine with this approach. Thanks.